### PR TITLE
mm/binance: Use filters and withdraw multiple

### PR DESCRIPTION
**Pull request by martonp**
The price and lot size filters were not being used, leading to trade
errors. The withdraw multiple was also not being used leading to withdraw
errors.

### CODE REVIEW

**Review by buck54321 — CHANGES_REQUESTED** [review]
I think this can be much simpler. Please review these two suggestions.

https://github.com/buck54321/dcrdex/compare/47d2deb...buck54321:dcrdex:preparsefilters

https://github.com/buck54321/dcrdex/compare/c6e8f5d...buck54321:dcrdex:withdraw-lots

File: client/mm/libxc/binance.go [hunk 1]
```diff
@@ -1256,12 +1319,21 @@ func (bnc *binance) generateRequest(ctx context.Context, method, endpoint string
...
 	req, err := http.NewRequestWithContext(ctx, method, fullURL, body)
 	if err != nil {
-		return nil, fmt.Errorf("NewRequestWithContext error: %w", err)
+		return fmt.Errorf("NewRequestWithContext error: %w", err)
 	}
 
 	req.Header = header
 
-	return req, nil
+	// bnc.log.Tracef("Sending request: %+v", req)
```
Comment (dev-warrior777) [inline client/mm/libxc/binance.go:1327]: Do you still need this?

File: client/mm/libxc/bntypes/types.go [hunk 2]
```diff
@@ -34,9 +57,9 @@ type NetworkInfo struct {
...
 	// UnLockConfirm           int     `json:"unLockConfirm"`
-	WithdrawEnable bool    `json:"withdrawEnable"`
-	WithdrawFee    float64 `json:"withdrawFee,string"`
-	// WithdrawIntegerMultiple float64 `json:"withdrawIntegerMultiple,string"`
+	WithdrawEnable          bool    `json:"withdrawEnable"`
+	WithdrawFee             float64 `json:"withdrawFee,string"`
+	WithdrawIntegerMultiple float64 `json:"withdrawIntegerMultiple,string"`
 	// WithdrawMax             float64 `json:"withdrawMax,string"`
 	WithdrawMin float64 `json:"withdrawMin,string"`
 	// SameAddress             bool    `json:"sameAddress"`
```
Comment (dev-warrior777) [inline client/mm/libxc/bntypes/types.go:65]: Maybe remove commented.

**Review by dev-warrior777 — APPROVED** [review]

**Comment by buck54321** [comment]
Needs a rebase, but look great.

File: client/mm/libxc/binance.go [hunk 3]
```diff
@@ -649,12 +658,44 @@ func (bnc *binance) getMarkets(ctx context.Context) (map[string]*bntypes.Market,
...
+		qui, _ := asset.UnitInfo(dexMkt.QuoteID)
+
+		var rateStepFound, lotSizeFound bool
+		for _, filter := range market.Filters {
+			if filter.Type == "PRICE_FILTER" {
+				rateStepFound = true
+				conv := float64(qui.Conventional.ConversionFactor) / float64(bui.Conventional.ConversionFactor) * calc.RateEncodingFactor
+				market.RateStep = uint64(math.Round(filter.TickSize * conv))
+				market.MinPrice = uint64(math.Round(filter.MinPrice * conv))
+				market.MaxPrice = uint64(math.Round(filter.MaxPrice * conv)) // Why was the conv outside of the math.Round?
```
Comment (buck54321) [inline client/mm/libxc/binance.go:679]: > // Why was the conv outside of the math.Round?

Not sure. Whadya expect me to test my code suggestions? 🤪

Comment (martonp) [reply client/mm/libxc/binance.go:679]: Lol that comment was kind of meant for myself.

File: client/mm/libxc/bntypes/types.go (hunk 2 above)
Comment (martonp) [reply client/mm/libxc/bntypes/types.go:65]: I think it's ok.. good to know what other fields there are without having to check the API.

**Review by JoeGruffins — APPROVED** [review]

**Pull request merged**